Foundation/include/Poco/DeflatingStream.h

```diff
@@ -104,7 +101,7 @@ class Foundation_API DeflatingIOS: public virtual std::ios
 	/// order of the stream buffer and base classes.
 {
 public:
-	DeflatingIOS(std::ostream& ostr, DeflatingStreamBuf::StreamType type = DeflatingStreamBuf::STREAM_ZLIB, int level = Z_DEFAULT_COMPRESSION);
+	DeflatingIOS(std::ostream& ostr, DeflatingStreamBuf::StreamType type = DeflatingStreamBuf::STREAM_ZLIB, int level = -1);
```

Comment:
Going from `Z_DEFAULT_COMPRESSION` to just `-1` is a loss of info for users. Can we add a `POCO_DEFAULT_COMPRESSION` const for this?

Reply:
There is an issue with other (non-default) compression levels whenever the user decided to go with non-default.

Shall Poco maintain a copy of compression values that are identical with the ones in zlib?

@andrewauclair , @teksturi